src/pipdeptree/_models/dag.py

```diff
@@ -36,19 +36,16 @@ class PackageDAG(Mapping[DistPackage, List[ReqPackage]]):
     """
 
     @classmethod
-    def from_pkgs(cls, pkgs: list[DistInfoDistribution]) -> PackageDAG:
-        dist_pkgs = [DistPackage(p) for p in pkgs]
+    def from_pkgs(cls, pkgs: list[Distribution]) -> PackageDAG:
+        dist_pkgs = [DistPackage(p) for p in pkgs if p.name]
         idx = {p.key: p for p in dist_pkgs}
         m: dict[DistPackage, list[ReqPackage]] = {}
         for p in dist_pkgs:
             reqs = []
             for r in p.requires():
-                # Requirement key is not sufficiently normalized in pkg_resources - apply additional normalization
-                d = idx.get(pep503_normalize(r.key))
-                # pip's _vendor.packaging.requirements.Requirement uses the exact casing of a dependency's name found in
-                # a project's build config, which is not ideal when rendering.
-                # See https://github.com/tox-dev/pipdeptree/issues/242
-                r.project_name = d.project_name if d is not None else r.project_name
+                if pep503_normalize(r.name) in [r_visited.key for r_visited in reqs]:
+                    continue
```

Comment:
Reimplement this in `Distribution.requires()`. To me it would make more since for this method to return all the direct dependencies.

Reply:
You mean in `DistPackage.requires()` ? If so, is the following code what we need?
```
def requires(self) -> list[Requirement]:
        req_list = []
        req_name_list = []
        if self._obj.requires:
            for r in self._obj.requires:
                req = Requirement(r)
                if req.name not in req_name_list:
                    req_list.append(req)
                    req_name_list.append(req.name)
        return req_list
```

Reply:
I add the filter of removing the requires with marker. I think it may work this time.